src/ml/neural_net/tf_compute_context.hpp

```diff
@@ -41,6 +41,13 @@ EXPORT class tf_compute_context: public compute_context {
       int n, int c_in, int h_in, int w_in, int c_out, int h_out, int w_out,
       const float_array_map& config, const float_array_map& weights) override;
 
+  std::unique_ptr<model_backend> create_drawing_classifier(
+      const variant_type &validation_set, 
```

Comment:
I also don't think validation set should be passed. I don't think variant type must be passed from C++ to python while making the model. It must be passed using the iterator while training and that's it.